# Phylogenomics Workflow Tutorial
### v 0.1.0

Hello there! This is the first iteration of a tutorial on how to use my phylogenomics pipeline (powered by Anvi'o and ITOL).

First, let's address a key question: why would you want to do this?

  - You have an insatiable desire to create phylogenetic trees
  - You have some unknown genomes that you want to get a taxonomic placement for
  - You want to pull out ribosomal genes from lots of genomes at once
  - You want to experience sassy error messages from Anvi'o

If any of those apply to you, read on! If not... do what you like, I suppose.

Please note that the following tutorial only applies to Unix-based systems. If you don't have a Unix-based system, consider getting one.

---

## Installation

There are two possible situations if you're reading this.

1. You are working at Pitt and have access to Microbiome01.

2. You don't have access to the Pitt cluster.

If the first one applies to you, either use my account or talk to Kelvin. (Emailing me would be preferable in this case if it's not running globally yet; I can give you the info you need to run this)

If the second one applies to you, we're going to do some installation.

If you don't have Python installed, go ahead and install that (instructions are distribution-specific, and I leave it in your capable hands).

First, go [here](https://www.anaconda.com/download/#linux) to download the Anaconda installer. Please ensure you download the version that fits the Python version you have.

Second, go to wherever you downloaded the file and perform the following:

```{r, engine='bash', install_conda}
bash Anaconda[VERSION NAME].sh
```

Then follow the helpful instructions. The wonderful thing about Anaconda is that this will work fine even if you're on a cluster where you don't have sudo access. Trying to install software without sudo can be aggravating, and conda tends to get past that particular hurdle very well.

This will create a local installation; if you're on a cluster with multiple users, this will not give them access to it!! In order to do so you will likely have to do a couple things, like:

 - modify the conda directory (which you will specify upon install) to be accessible to all users if necessary.
 - Take the line generated by conda in your .bashrc (in your home directory; try using 'ls -a' to find it) and have other users put it in their .bashrc files.

Congratulations! From this point on installing things should be easy. Make sure you can run conda:

```{r, engine='bash', check_conda}
conda --version
```

If everything's good, proceed. Otherwise... try again or contact me with your error. Google is also a reliable friend in instances like these, and you will likely be able to solve this most quickly by googling it. Please try that before contacting me or I will tell you to google it.

### Installing Anvi'o
Watch this.

```{r, engine='bash', install_anvio}
conda create -n anvio4 -c bioconda -c conda-forge python=3 anvio=4
```

Boom.

### Installing iqtree

And again, bask in the glory of easy software installation.

```{r, engine='bash', install_anvio}
conda install -c bioconda iqtree
```

Ta-dah!
